provide/Services/MessageProtocolParser.cs

```diff
@@ -0,0 +1,89 @@
+using System.IO;
+using System.Text;
+
+public class Message
+{
+    public string OpCode { get => "BLINE"; } // up to 40 bits
+    public string Sender { get; set; } // up to 336 bits
+    public string Recipient { get; set; } // up to 336 bits
+    public string Shield { get; set; } // up to 336 bits
+    public string Identifier { get; set; } // up to 288 bits (i.e., UUIDv4 circuit/workflow identifier)
+    public string Signature { get; set; } // 512 bits
+    public string Type { get => "1"; }  // 1 bit
+    public byte[] Payload { get; set; } // arbitrary length
+}
+
+public static class MessageProtocolParser
+{
+    const int MessageReservedBitsLength = 512;
+    public static byte[] MarshalProtocolMessage(Message msg)
+    {
+        var reservedBits = new byte[MessageReservedBitsLength / 8];
+        var buffer = new byte[5 + 42 + 42 + 42 + 36 + 64 + 1 + reservedBits.Length + msg.Payload.Length];
+
+        using (MemoryStream ms = new MemoryStream(buffer))
+        {
+            using(BinaryWriter bw = new BinaryWriter(ms, Encoding.UTF8))
+            {
+                bw.Write(msg.OpCode.ToCharArray());
```

Comment:
converting string to char array because binary writer in case of strings adds length of string as first in sequence of bytes